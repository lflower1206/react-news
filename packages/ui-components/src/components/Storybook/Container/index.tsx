import React, { FC } from 'react'
import { styled } from '@storybook/theming'
import { ThemeProvider } from 'styled-components'

import theme from '../../../theme'

const ContainerDiv = styled.div`
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto,
    Microsoft JhengHei, Helvetica Neue, Helvetica, Arial, sans-serif;

  padding: 16px 24px;
`

const Container: FC = props => {
  return (
    <ThemeProvider theme={theme}>
      <ContainerDiv>{props.children}</ContainerDiv>
    </ThemeProvider>
  )
}

export default Container
