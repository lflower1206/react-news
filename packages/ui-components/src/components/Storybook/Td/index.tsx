import { styled } from '@storybook/theming'

const Td = styled.td`
  padding: 24px 0;
  border-bottom: 1px solid rgba(0, 0, 0, 0.08);

  &:first-child {
    padding-right: 120px;
  }
`

export default Td
