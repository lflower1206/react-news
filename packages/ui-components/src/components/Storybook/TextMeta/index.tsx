import { styled } from '@storybook/theming'

const TextMeta = styled.div`
  color: rgba(85, 85, 85, 0.6);
  font-size: 10px;
  line-height: 1.2;
`

export default TextMeta
