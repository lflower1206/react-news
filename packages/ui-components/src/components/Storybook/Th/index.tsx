// import { styled } from '@storybook/theming'

import styled, { StyledComponent } from 'styled-components'
import { layout } from 'styled-system'

export interface IThProps {
  width?: number
}
const Th: StyledComponent<'th', {}, IThProps> = styled.th`
  padding: 8px 0;
  border-bottom: 1px solid rgba(0, 0, 0, 0.16);
  ${layout}
`

export default Th
