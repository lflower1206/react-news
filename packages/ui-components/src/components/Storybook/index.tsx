export { default as Container } from './Container'
export { default as TextMeta } from './TextMeta'
export { default as Td } from './Td'
export { default as Th } from './Th'
