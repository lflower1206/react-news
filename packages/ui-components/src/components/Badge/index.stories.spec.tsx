import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import { badge as BadgeStories } from './index.stories'
import { Container } from '../Storybook'

describe('Badge', () => {
  it('should render Badge stories', () => {
    const { asFragment } = render(
      <Container>
        <BadgeStories />
      </Container>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
