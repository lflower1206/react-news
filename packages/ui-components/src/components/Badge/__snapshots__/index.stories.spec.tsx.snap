// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render Badge stories 1`] = `
<DocumentFragment>
  .c1 {
  border-top-left-radius: 4px;
  border-bottom-right-radius: 4px;
  background-color: #000;
  color: #fff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 2px 6px;
  font-size: 12px;
}

.c2 {
  border-radius: 4px;
  background-color: #000;
  color: #fff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 2px 6px;
  font-size: 12px;
}

.c0 {
  padding-bottom: 16px;
}

<div
    class="css-lii6sj"
  >
    <ul>
      <li
        class="c0"
      >
        <div
          class="c1"
          color="#fff"
        >
          Dimmed Badge
        </div>
        <div
          class="css-nychs"
        >
          Rectangle Dimmed
        </div>
      </li>
      <li
        class="c0"
      >
        <div
          class="c2"
          color="#fff"
        >
          Solid Badge
        </div>
        <div
          class="css-nychs"
        >
          Rectangle Solid
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
